api/v1alpha1/cozystackplatformconfiguration_types.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2025 The Cozystack Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	sourcev1 "github.com/fluxcd/source-controller/api/v1"
+)
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Cluster
+
+// CozystackPlatformConfiguration is the Schema for the cozystackplatformconfigurations API
+type CozystackPlatformConfiguration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec CozystackPlatformConfigurationSpec `json:"spec,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// CozystackPlatformConfigurationList contains a list of CozystackPlatformConfigurations
+type CozystackPlatformConfigurationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []CozystackPlatformConfiguration `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&CozystackPlatformConfiguration{}, &CozystackPlatformConfigurationList{})
+}
+
+// CozystackPlatformConfigurationSpec defines the desired state of CozystackPlatformConfiguration
+type CozystackPlatformConfigurationSpec struct {
+	// Source configuration for GitRepository
+	// +required
+	Source CozystackPlatformConfigurationSource `json:"source"`
+
+	// Chart configuration for HelmRelease
+	// +required
+	Chart CozystackPlatformConfigurationChart `json:"chart"`
+
+	// Values to pass to HelmRelease
+	// +optional
+	Values *apiextensionsv1.JSON `json:"values,omitempty"`
+}
+
+// CozystackPlatformConfigurationSource defines the source configuration for GitRepository
+// Reuses Flux GitRepositorySpec fields
+type CozystackPlatformConfigurationSource struct {
+	// URL of the Git repository
+	// +required
+	URL string `json:"url"`
+
+	// Git repository reference (branch, tag, semver, commit)
+	// +optional
+	Ref *sourcev1.GitRepositoryRef `json:"ref,omitempty"`
```

Comment:
If you're referencing `sourcev1.GitRepositoryRef`, why don't you just inline the entire `GitRepositorySpec` from flux as is instead of copying it field by field?

Reply:
Now this resource just have simple SourceRef